mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -1256,4 +1256,27 @@ def Rock_InWarpTransposeOp :
     # !cast<string>(swizzleGroupSize) # ";\n";
 }
 
+def Rock_ThreadwiseTransposeOp :
+    Rock_Op<"threadwise_transpose">,
+    AllElementTypesMatch<["source", "dest"]>,
+    Arguments<(ins
+      Arg<MemRefOf<SupportedMemoryElems>, "source view", [MemRead]>:$source,
+      Arg<MemRefOf<SupportedMemoryElems>, "detination view", [MemWrite]>:$dest,
+      I64Attr : $kpack,
+      UnitAttr:$forceUnroll,
+      UnitAttr:$useIndexDiffs
+    )> {
+  let summary = "Threadwise register-to-register transpose";
+
+  let description = [{
+    `rock.threadwise_transpose` accepts a source view and a destination view and copy
+    data from source to destination.
```

Comment:
This is a bit misleading description for what it does.
The current lowering only works for a very specific dimensionality in the upper views that needs to be K x D.
I think we need a op verifier too to ascertain this.